tests/test_cli.py

```diff
@@ -20,6 +25,51 @@ def context():
     return ctx().obj["context"]
 
 
+mock_gui_main = """from .app import show
```

Comment:
So we're on the same page, what the `gui` argument is meant to do is call on the module you give it, and assumes it will run something - in most cases, show the GUI. For testing purposes, it's enough to simply test for whether it runs or not.

For data, the important bit is that the resulting call to the module you give it is given access to it. From an actual GUI, the data would be available to the plug-ins that run within that GUI. We can test that by simply calling `util.publish()` and inspecting the resulting `context`.